repos/system_upgrade/common/actors/initramfs/upgradeinitramfsgenerator/libraries/upgradeinitramfsgenerator.py

```diff
@@ -188,14 +207,24 @@ def copy_boot_files(context):
     curr_arch = api.current_actor().configuration.architecture
     kernel = 'vmlinuz-upgrade.{}'.format(curr_arch)
     initram = 'initramfs-upgrade.{}.img'.format(curr_arch)
+
+    fips_info = next(api.consume(FIPSInfo), None)
+    kernel_hmac = '.{0}.hmac'.format(kernel)
+    kernel_hmac_path = os.path.join('/boot', kernel_hmac) if fips_info.is_enabled else None
+
     content = BootContent(
         kernel_path=os.path.join('/boot', kernel),
-        initram_path=os.path.join('/boot', initram)
+        initram_path=os.path.join('/boot', initram),
+        kernel_hmac_path=kernel_hmac_path
     )
 
     context.copy_from(os.path.join('/artifacts', kernel), content.kernel_path)
     context.copy_from(os.path.join('/artifacts', initram), content.initram_path)
 
+    if fips_info.is_enabled:
+        kernel_hmac_path = context.full_path(os.path.join('/artifacts', kernel_hmac))
+        create_upgrade_hmac_from_target_hmac(kernel_hmac_path, content.kernel_hmac_path, kernel)
```

Comment:
@pirat89 Based on your experience from the 6 > 7 upgrades, could you please take a look if there might be any issue around here?

Reply:
This seems ok to me :+1: